flink-scala/src/main/scala/org/apache/flink/api/scala/ExecutionEnvironment.scala

```diff
@@ -247,16 +252,27 @@ class ExecutionEnvironment(javaEnv: JavaEnv) {
       inputFormat.enableQuotedStringParsing(quoteCharacter);
     }
 
-    val classes: Array[Class[_]] = new Array[Class[_]](typeInfo.getArity)
+    val classesBuf: ArrayBuffer[Class[_]] = new ArrayBuffer[Class[_]]
     for (i <- 0 until typeInfo.getArity) {
-      classes(i) = typeInfo.getTypeAt(i).getTypeClass
+      typeInfo match {
+        case info: TupleTypeInfoBase[T] => classesBuf += info.getTypeAt(i).getTypeClass()
+        case info: PojoTypeInfo[T] =>
+          if (includedFields == null || includedFields.indexOf(i) != -1) {
```

Comment:
Just add all POJO fields without checking against includedFields. IncludedFields refer to the fields of the CSV file not to the fields of the POJO.
The later check ensures that the number of read fields and the number of POJO/Tuple fields is equal.